crypto/BRCryptoPayment.c

```diff
@@ -234,46 +243,48 @@ static BRCryptoPaymentProtocolRequest
-
-        BRCryptoUnit feeUnit = cryptoNetworkGetUnitAsBase (builder->cryptoNetwork, builder->cryptoCurrency);
-        BRCryptoAmount feeAmount = cryptoAmountCreateDouble (builder->feePerByte, feeUnit);
-        protoReq->u.btc.requiredFee = cryptoNetworkFeeCreate (0, feeAmount, feeUnit);
-        cryptoAmountGive (feeAmount);
-        cryptoUnitGive (feeUnit);
-
-        protoReq->u.btc.request = request;
-
-    } else {
-        BRPaymentProtocolDetailsFree (details);
+    if ((BLOCK_CHAIN_TYPE_BTC == cryptoNetworkGetType (builder->cryptoNetwork)) &&
+         cryptoNetworkHasCurrency(builder->cryptoNetwork, builder->cryptoCurrency) &&
+         0 != array_count (builder->outputs) && 0 != builder->outputs[0].amount && 0 != builder->outputs[0].scriptLen) {
+
+        BRPaymentProtocolDetails *details = BRPaymentProtocolDetailsNew (builder->network,
+                                                                         builder->outputs,
+                                                                         array_count (builder->outputs),
+                                                                         builder->time,
+                                                                         builder->expires,
+                                                                         builder->memo,
+                                                                         builder->paymentURL,
+                                                                         builder->merchantData,
+                                                                         builder->merchantDataLen);
+
+        BRPaymentProtocolRequest *request = BRPaymentProtocolRequestNew (1,
+                                                                        "none",
+                                                                        NULL,
+                                                                        0,
+                                                                        details,
+                                                                        NULL,
+                                                                        0);
+
+        if (NULL != request) {
+            protoReq = calloc (1, sizeof (struct BRCryptoPaymentProtocolRequestRecord));
+            protoReq->ref = CRYPTO_REF_ASSIGN(cryptoPaymentProtocolRequestRelease);
+            protoReq->type = CRYPTO_PAYMENT_PROTOCOL_TYPE_BITPAY;
+
+            protoReq->cryptoNetwork = cryptoNetworkTake (builder->cryptoNetwork);
+            protoReq->cryptoCurrency = cryptoCurrencyTake (builder->cryptoCurrency);
+
+            BRCryptoUnit feeUnit = cryptoNetworkGetUnitAsBase (builder->cryptoNetwork, builder->cryptoCurrency);
+            BRCryptoAmount feeAmount = cryptoAmountCreateDouble (builder->feePerByte, feeUnit);
+            protoReq->u.btc.requiredFee = cryptoNetworkFeeCreate (0, feeAmount, feeUnit);
```

Comment:
You are creating an essentially fake NetworkFee here - with a confirmationTime of '0'?

Reply:
Not fake, from the BitPay request. Fees have a time but I can't possible know the time as that isn't provided by the bitpay protocol.

Reply:
Yes, surely a fee is required - but in this case it isn't related to the network's 'time to confirm'.